libsrc/apple2/acc_detect_fastchip.s

```diff
@@ -0,0 +1,46 @@
+;
+; Marco van den Heuvel, 2018-05-04
+;
+
+; unsigned char detect_fastchip (void);
+;
+;/* Check for the presence of the Apple IIe Fast Chip slot card.
+; *
+; * Possible return values:
+; * 0x00  : Fast Chip slot card not present
+; * 0x01  : Fast Chip slot card present
+; */
+
+        .export         _detect_fastchip
+        .import         _get_ostype
+
+        .include        "accelerator.inc"
+
+_detect_fastchip:
+        jsr     _get_ostype
+        ldx     #$00
+        cmp     #$30                ; Apple IIe
+        beq     check_slot
+        cmp     #$31                ; Apple IIe enhanced
+        beq     check_slot
+        cmp     #$40                ; Apple IIe Option Card
+        bne     not_found
+
+check_slot:
```

Comment:
`check_slot` is a bad name (tm) here -  and likely the reason why @greg-king5 asked a "wrong" question below.

The usual way slot cards work in the Apple II is that each slot has an individual range of I/O addresses assigned. These ranges are $C080-$C08F for slot 0, $C090-$09F for slot 1 etc. So to "check slots" for a card would mean to access all the different I/O address ranges in a loop. But here there's neither a loop nor access to $C080-$C0FF.

The FastChipIIe works differently than most slot cards. It doesn't rely on the logic decribed above but rather fully monitors the 6502 bus activities. Therefore it
- places its I/O ports in a slot-independent location ($C06x)
- can emulate a RamFactor card in a slot it isn't physically placed in
- ...

Reply:
I understand, the CPU on that card is in control and therefor controls the memory map available to the CPU.

Reply:
What I wrote doesn‘t relate to the fact that the card in question is an accelerator with its own CPU. Rather the Apple II „offers“ some built-in address decoding to slot cards to
- allow them to decode only 4 address bits
- allow them to co-operate without address conflicts

However, the card in question doesn‘t make use of that „offer“.